amp/WEB-INF/src/org/digijava/module/aim/util/FeaturesUtil.java

```diff
@@ -1254,24 +1254,32 @@ public static boolean deleteTemplateVisibility(Long id) {
             AmpTemplatesVisibility ft = new AmpTemplatesVisibility();
             ft = (AmpTemplatesVisibility) hbsession.load(AmpTemplatesVisibility.class,id);
 
-            for (AmpFieldsVisibility f:ft.getFields()) {
-                f.getTemplates().remove(ft);
-            }
-            for (AmpFeaturesVisibility f:ft.getFeatures()) {
-                f.getTemplates().remove(ft);
-            }
-            for (Iterator it = ft.getItems().iterator(); it.hasNext();) {
-                AmpModulesVisibility f = (AmpModulesVisibility) it.next();
-                f.getTemplates().remove(ft);
+            if (ft != null) {
+                for (AmpFieldsVisibility f : ft.getFields()) {
+                    f.getTemplates().remove(ft);
+                }
+                for (AmpFeaturesVisibility f : ft.getFeatures()) {
+                    f.getTemplates().remove(ft);
+                }
+                for (Iterator it = ft.getItems().iterator(); it.hasNext(); ) {
+                    AmpModulesVisibility f = (AmpModulesVisibility) it.next();
+                    f.getTemplates().remove(ft);
+                }
+
+                hbsession.delete(ft);
+                hbsession.getTransaction().commit();
             }
-            ft.clearFields();
-            ft.getFeatures().clear();
-            ft.getItems().clear();
-            hbsession.delete(ft);
-            
         }
         catch (Exception ex) {
+            if (hbsession != null && hbsession.getTransaction().isActive()) {
```

Comment:
we should not explicitly commit or rollback since that is handled by the hibernate filter

Reply:
Okay noted